ansible/roles/vm_set/tasks/start.yml

```diff
@@ -12,29 +17,55 @@
     - "{{ root_path }}/images"
     - "{{ root_path }}/disks"
 
-- name: Check hdd image
-  stat: path={{ root_path }}/images/{{ hdd_image_filename }}
-  register: hdd_stat
+- block:
+
+    - name: Check hdd image
+      stat: path={{ root_path }}/images/{{ hdd_image_filename }}
+      register: hdd_stat
+
+    - name: Fail if there are no hdd image and skip image downloading is active
+      fail: msg="Please put {{ hdd_image_filename }} to {{ root_path }}/images"
+      when: not hdd_stat.stat.exists and skip_image_downloading
+
+    - name: Download hdd image
+      get_url: url="{{ vm_images_url }}/{{ hdd_image_filename }}?{{ vmimage_saskey }}" dest="{{ root_path }}/images/{{ hdd_image_filename }}"
+      when: not hdd_stat.stat.exists and not skip_image_downloading
+
+    - name: Check cd image
+      stat: path={{ root_path }}/images/{{ cd_image_filename }}
+      register: cd_stat
+
+    - name: Fail if there are no cd image and skip image downloading is active
+      fail: msg="Please put {{ cd_image_filename }} to {{ root_path }}/images"
+      when: not cd_stat.stat.exists and skip_image_downloading
+
+    - name: Download cd image
+      get_url: url="{{ vm_images_url }}/{{ cd_image_filename }}?{{ cdimage_saskey }}" dest="{{ root_path }}/images/{{ cd_image_filename }}"
+      when: not cd_stat.stat.exists and not skip_image_downloading
+
+    - set_fact:
+        src_image_name: "{{ hdd_image_filename }}"
+
+  when: (vm_type | lower) == "veos"
+
+- block:
 
-- name: Fail if there are no hdd image and skip image downloading is active
-  fail: msg="Please put {{ hdd_image_filename }} to {{ root_path }}/images"
-  when: not hdd_stat.stat.exists and skip_image_downloading
+    - name: Check SONiC image
+      stat: path={{ root_path }}/images/{{ sonic_image_filename }}
+      register: img_stat
 
-- name: Download hdd image
-  get_url: url="{{ vm_images_url }}/{{ hdd_image_filename }}?{{ vmimage_saskey }}" dest="{{ root_path }}/images/{{ hdd_image_filename }}"
-  when: not hdd_stat.stat.exists and not skip_image_downloading
+    - name: Fail if there are no SONiC image and skip image downloading is active
+      fail: msg="Please put {{ sonic_image_filename }} to {{ root_path }}/images"
+      when: not img_stat.stat.exists and skip_image_downloading
```

Comment:
If we do not support downloading sonic-vs.img for now, maybe it is better to only check if the image file exists and do not check flag `skip_image_downloading`.